internal/pkg/service/buffer/worker/distribution/listener.go

```diff
@@ -50,10 +54,16 @@ func newListeners(proc *servicectx.Process, clock clock.Clock, logger log.Logger
 	go func() {
 		defer wg.Done()
 
-		// Listeners are not triggered immediately on change,
-		// but all events within the groupInterval are processed at once.
-		triggerTicker := clock.Ticker(config.eventsGroupInterval)
-		defer triggerTicker.Stop()
+		var tickerC <-chan time.Time
+		if config.eventsGroupInterval > 0 {
+			// Listeners are not triggered immediately on change,
+			// but all events within the groupInterval are processed at once.
+			triggerTicker := clock.Ticker(config.eventsGroupInterval)
+			defer triggerTicker.Stop()
+		} else {
+			// No grouping interval, trigger is called immediately, see Notify method.
+			tickerC = make(chan time.Time)
+		}
```

Comment:
V `distribution` package mame taku feature `eventsGrouping`.
To znamena, ze vsetky zmeny v distribucii za napr. `5s` sa spracuju naraz.
Znizi to pocet zbytocneho opakovania prace/callbacks, ktore sa musia zavolat pri zmene distribucie.
Ak v kratkom case po sebe konci/zacina viacero workers, napr. pri deploymente.